code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm

```diff
@@ -0,0 +1,455 @@
+	. += waste_cap_appearance
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/atmos_init(list/node_connects)
+	for(var/obj/machinery/atmospherics/target in get_step(src, dir))
+		if(connection_check(target, 4) && !nodes[1])
+			nodes[1] = target // Distro
+		if(connection_check(target, 2) && !nodes[2])
+			nodes[2] = target // Waste
+	update_appearance()
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		can_unwrench = FALSE
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/post_machine_initialize()
+	. = ..()
+	set_links()
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/New()
+	. = ..()
+	var/datum/gas_mixture/distro_air = airs[1]
+	var/datum/gas_mixture/waste_air = airs[2]
+	distro_air.volume = 1000
+	waste_air.volume = 1000
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/on_deconstruction(disassembled)
+	. = ..()
+	if(cycling_set_up)
+		break_all_links()
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/can_unwrench(mob/user)
+	. = ..()
+	if(!.)
+		to_chat(user, span_warning("You cannot unwrench [src], it is secured firmly in place!"))
+		return FALSE
+	if(. && on)
+		to_chat(user, span_warning("You cannot unwrench [src], wait for the cycle completion!"))
+		return FALSE
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/process_atmos()
+	if(!on)
+		return
+
+	if(!powered())
+		stop_cycle("No power. Cycle aborted.", unbolt_only = TRUE)
+		return //Couldn't complete the cycle due to power outage
+
+	var/turf/location = get_turf(loc)
+	if(isclosedturf(location))
+		return
+
+	if(COOLDOWN_FINISHED(src, check_turfs_cooldown))
+		check_turfs()
+		COOLDOWN_START(src, check_turfs_cooldown, 2 SECONDS)
+
+	if(world.time - cycle_start_time > cycle_timeout)
+		stop_cycle("Cycling timed out, bolts unlocked.", unbolt_only = TRUE)
+		return //Couldn't complete the cycle before timeout
+
+	var/datum/gas_mixture/distro_air = airs[1]
+	var/datum/gas_mixture/tile_air = loc.return_air()
+	var/tile_air_pressure = tile_air.return_pressure()
+
+	if(pump_direction == ATMOS_DIRECTION_RELEASING) //distro node -> tile
+		var/pressure_delta = cycle_pressure_target - tile_air_pressure
+		if(pressure_delta <= allowed_pressure_error && stop_cycle("Pressurization complete."))
+			return //Internal target pressure reached
+
+		var/available_moles = distro_air.total_moles()
+		var/total_tiles = adjacent_turfs.len + 1
+		var/split_moles = QUANTIZE(available_moles / total_tiles)
+
+		fill_tile(loc, split_moles, pressure_delta)
+		for(var/turf/tile in adjacent_turfs)
+			fill_tile(tile, split_moles, pressure_delta)
+	else //tile -> waste node
+		var/pressure_delta = tile_air_pressure - cycle_pressure_target
+		if(pressure_delta <= allowed_pressure_error && stop_cycle("Decompression complete."))
+			return //External target pressure reached
+
+		siphon_tile(loc)
+		for(var/turf/tile in adjacent_turfs)
```

Comment:
```suggestion
		for(var/turf/tile as anything in adjacent_turfs)
```

Reply:
Why? I'm just passing it to a proc. In fact, even `/turf/` isn't needed here.